11_FFTBloom/app_resources/fft_convolve_ifft.hlsl

```diff
@@ -0,0 +1,204 @@
+{
+	return pushConstants.rowMajorBufferAddress + channel * glsl::gl_NumWorkGroups().x * pushConstants.dataElementCount * sizeof(complex_t<scalar_t>);
+}
+
+
+// ------------------------------------------ SECOND AXIS FFT + CONVOLUTION + IFFT -------------------------------------------------------------
+
+// This time each Workgroup will compute the FFT along a horizontal line (fixed y for the whole Workgroup). We get the y coordinate for the
+// row a workgroup is working on via `gl_WorkGroupID().x`. We have to keep this in mind: What's stored as the first row is actually `Z + iN`, 
+// where `Z` is the actual 0th row and `N` is the Nyquist row (the one with index FFT_LENGTH / 2). Those are packed together
+// so they need to be unpacked properly after FFT like we did earlier.
+
+struct PreloadedSecondAxisAccessor : PreloadedAccessorBase<ELEMENTS_PER_THREAD, _NBL_HLSL_WORKGROUP_SIZE_, scalar_t>
+{
+	int32_t mirrorWrap(int32_t paddedCoordinate)
+	{
+		const int32_t negMask = paddedCoordinate >> 31u;
+		const int32_t d = ((paddedCoordinate ^ negMask) / pushConstants.dataElementCount) ^ negMask;
+		paddedCoordinate = paddedCoordinate - d * pushConstants.dataElementCount;
+		const int32_t flip = d & 0x1;
+		return (1 - flip) * paddedCoordinate + flip * (pushConstants.dataElementCount - 1 - paddedCoordinate); //lerping is a float op
+	}
+
+	void preload(uint32_t channel)
+	{
+		// Set up accessor to point at channel offsets
+		bothBuffersAccessor = DoubleLegacyBdaAccessor<complex_t<scalar_t> >::create(getColMajorChannelStartAddress(channel), getRowMajorChannelStartAddress(channel));
+		const uint32_t padding = uint32_t(FFT_LENGTH - pushConstants.dataElementCount) >> 1;
+
+		for (uint32_t elementIndex = 0; elementIndex < ELEMENTS_PER_THREAD; elementIndex++)
+		{
+			const uint32_t index = _NBL_HLSL_WORKGROUP_SIZE_ * elementIndex | workgroup::SubgroupContiguousIndex();
+			const int32_t paddedIndex = index - int32_t(padding);
+			const int32_t wrappedIndex = mirrorWrap(paddedIndex);
+			preloaded[elementIndex] = bothBuffersAccessor.get(colMajorOffset(wrappedIndex, glsl::gl_WorkGroupID().x));
+		}
+	}
+
+	// Each element on this row is Nabla-ordered. So the element at `x' = index, y' = gl_WorkGroupID().x` that we're operating on is actually the element at
+	// `x = F(index), y = bitreverse(gl_WorkGroupID().x)` (with the bitreversal done as an N-1 bit number, for `N = log2(FFT_LENGTH)` *of the first axist FFT*)
+	template<typename SharedmemAdaptor>
+	void convolve(uint32_t channel, SharedmemAdaptor sharedmemAdaptor)
+	{
+		// Remember first row holds Z + iN
+		if (!glsl::gl_WorkGroupID().x)
+		{
+			for (uint32_t localElementIndex = 0; localElementIndex < ELEMENTS_PER_THREAD; localElementIndex += 2)
+			{
+				complex_t<scalar_t> zero = preloaded[localElementIndex];
+				complex_t<scalar_t> nyquist = getDFTMirror<SharedmemAdaptor>(localElementIndex, sharedmemAdaptor);
+
+				workgroup::fft::unpack<scalar_t>(zero, nyquist);
```

Comment:
btw nitpick, maybe `unpack` shouldn't be in `workgroup::fft`, and just `fft` instead?

The reason is that as a developer, you expect anything in `workgroup::` namespace to imply using shared memory or barriers (so intuition should be to avoid divergent flow as if for a `barrier()`), and `subgroup::` to imply subgroup ops

Now obvioulsy the `unpack` function seems quite general purpose